site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -10173,6 +10233,38 @@ private void actionForTemplate(String direction, int index,
 		}
 
 	}// actionFor Template
+
+	/**
+	 * Responsible for hard deleting a temporary duplicate site. This method
+	 * is called during the duplication routine if an error is produced after
+	 * the duplicate site has been created (some of the functions require the
+	 * site to exist to perform some of the duplication processes, which can
+	 * sometimes result in errors that the user needs to rectify).
+	 *
+	 * @param tempDupSite the temporary duplicate site to be hard deleted in entirety
+	 */
+	private void deleteTempDupSiteOnError(Site tempDupSite) {
+		// We need to remove the duplicated site because errors were thrown
+		try {
+			SecurityAdvisor yesMan = new SecurityAdvisor() {
+				public SecurityAdvice isAllowed(String userId, String function, String reference) {
+					return SecurityAdvice.ALLOWED;
+				}
+			};
+			SecurityService.pushAdvisor(yesMan);
+
+			// Hard delete. Call upon all implementing services to hard delete their own content
+			doHardDelete(tempDupSite.getId());
+			// The service never deletes the site unless its already softly deleted
+			tempDupSite.setSoftlyDeleted(true);
+
+			// Now hard delete the site
+			SiteService.removeSite(tempDupSite);
+		} catch (Exception e) {
```

Comment:
should at least log one line here

Reply:
ditto here, existing code that was just moved.